include/fiction/algorithms/physical_design/hexagonalization.hpp

```diff
@@ -47,6 +107,29 @@ struct hexagonalization_stats
 namespace detail
 {
 
+/**
+ * This struct encapsulates a routing objective that specifies the source and target coordinates,
+ * along with a flag indicating whether the primary input was the first fanin for the corresponding fanout.
+ * @tparam HexLyt type of the hexagonal layout.
+ */
+template <typename HexLyt>
+struct extended_routing_objective
+{
+    /**
+     * The coordinate of the moved primary input.
+     */
+    tile<HexLyt> source;
+    /**
+     * The coordinate of the fanout node.
+     */
+    tile<HexLyt> target;
+    /**
+     * Flag that is set to true if the primary input was the first fanin; this indicates that the
+     * fanin signals need to be reordered.
+     */
+    bool update_first_fanin = false;
+};
```

Comment:
It's called `extended_routing_objective` but doesn't extend `routing_objective` :smile: I'd suggest using
```c++
template <typename HexLyt>
struct extended_routing_objective : public routing_objective<HexLyt>
```
Additionally, I'd try to find a more meaningful name because `extended_` could mean anything and thus doesn't convey an intuition.